### PR TITLE
automatic pitch

### PR DESCRIPTION
**Pull request by NitroTF2**
automatic pitch changer, might add some edge logic to it later.

### CODE REVIEW

**Comment by NitroTF2** [comment]
> kid named poo

fuck you

**Review by lnx00 — CHANGES_REQUESTED** [review]
The if statements are invalid and do not actually compare the class numbers. This means that most values will always be true.

File: Fedoraware/TeamFortress2/TeamFortress2/Features/AntiHack/AntiAim.cpp [hunk 1]
```diff
@@ -246,6 +246,38 @@ std::pair<float, float> CAntiAim::GetAnglePairPitch(int nIndex) {
 		G::FakeViewAngles.x = FakeDown ? 89.f : -89.f;
 		break;
 	}
+	case 11:
+	{
+		if (pLocal->GetClassNum() == CLASS_SCOUT && CLASS_SPY && CLASS_PYRO) // no neeed to change their pitch
```
Comment (johnathon-walnut) [inline Fedoraware/TeamFortress2/TeamFortress2/Features/AntiHack/AntiAim.cpp:251]: Are you retarded

File: Fedoraware/TeamFortress2/TeamFortress2/Features/AntiHack/AntiAim.cpp [hunk 2]
```diff
@@ -246,6 +246,38 @@ std::pair<float, float> CAntiAim::GetAnglePairPitch(int nIndex) {
...
+			{
+				retnAngles.first = 89.f;
+				retnAngles.second = 89.f;
+				break;
+			}
+			if (pLocal->GetClassNum() == CLASS_HEAVY && CLASS_DEMOMAN && CLASS_SOLDIER && CLASS_ENGINEER && CLASS_MEDIC && CLASS_SNIPER && pWeapon->GetSlot() == SLOT_PRIMARY) {
```
Comment (johnathon-walnut) [inline Fedoraware/TeamFortress2/TeamFortress2/Features/AntiHack/AntiAim.cpp:257]: This doesn't work

File: Fedoraware/TeamFortress2/TeamFortress2/Features/AntiHack/AntiAim.cpp [hunk 3]
```diff
@@ -246,6 +246,38 @@ std::pair<float, float> CAntiAim::GetAnglePairPitch(int nIndex) {
...
+				retnAngles.first = -89.f;
+				retnAngles.second = -89.f;
+				break;
+			}
+			if (pLocal->GetClassNum() == CLASS_HEAVY && CLASS_DEMOMAN && CLASS_SOLDIER && CLASS_ENGINEER && CLASS_MEDIC && CLASS_SNIPER && pWeapon->GetSlot() == SLOT_SECONDARY)
```
Comment (johnathon-walnut) [inline Fedoraware/TeamFortress2/TeamFortress2/Features/AntiHack/AntiAim.cpp:262]: Why are you like this

File: Fedoraware/TeamFortress2/TeamFortress2/Features/AntiHack/AntiAim.cpp [hunk 4]
```diff
@@ -246,6 +246,38 @@ std::pair<float, float> CAntiAim::GetAnglePairPitch(int nIndex) {
...
+			{
+				retnAngles.first = 89.f;
+				retnAngles.second = 89.f;
+				break;
+			}
+			if (pLocal->GetClassNum() == CLASS_HEAVY && CLASS_DEMOMAN && CLASS_SOLDIER && CLASS_ENGINEER && CLASS_MEDIC && CLASS_SNIPER && pWeapon->GetSlot() == SLOT_MELEE)
```
Comment (johnathon-walnut) [inline Fedoraware/TeamFortress2/TeamFortress2/Features/AntiHack/AntiAim.cpp:268]: This still doesn't work

File: Fedoraware/TeamFortress2/TeamFortress2/Features/AntiHack/AntiAim.cpp [hunk 5]
```diff
@@ -246,6 +246,38 @@ std::pair<float, float> CAntiAim::GetAnglePairPitch(int nIndex) {
...
+			{
+				retnAngles.first = 89.f;
+				retnAngles.second = 89.f;
+				break;
+			}
+			if (pLocal->GetClassNum() == CLASS_MEDIC && pWeapon->GetSlot() == SLOT_SECONDARY && (speed < 2.0f)) // if you move it changes pitch
```
Comment (johnathon-walnut) [inline Fedoraware/TeamFortress2/TeamFortress2/Features/AntiHack/AntiAim.cpp:274]: This one actually does good job

File: Fedoraware/TeamFortress2/TeamFortress2/Features/AntiHack/AntiAim.cpp (hunk 2 above)
Comment (NitroTF2) [reply Fedoraware/TeamFortress2/TeamFortress2/Features/AntiHack/AntiAim.cpp:257]: FIXING

**Comment by NitroTF2** [comment]
FIXING IT

**Comment by lnx00** [comment]
wtf

**Comment by lnx00** [comment]
https://en.wikipedia.org/wiki/Boolean_algebra

**Comment by johnathon-walnut** [comment]
@NitroTF2 can you check to make sure this is the same logic you initially proposed

**Comment by NitroTF2** [comment]
> @NitroTF2 can you check to make sure this is the same logic you initially proposed

yes if the vel speed is above 2 then change it

**Comment by NitroTF2** [comment]
but you don't change on secondary slot or melee...

**Comment by johnathon-walnut** [comment]
neither do you, since the angles are the same as everything else why check?

**Comment by NitroTF2** [comment]
> neither do you, since the angles are the same as everything else why check?

cuz works best to medic and i do change angle

Uploading 2022-08-10 23-08-30.mp4…

s

**Comment by NitroTF2** [comment]
@johnathon-walnut

**Comment by johnathon-walnut** [comment]
> > neither do you, since the angles are the same as everything else why check?
> 
> cuz works best to medic and I do change angle
> 
> based on my logics
> 2022-08-10.23-08-30.mp4

tard, my point is that there's no point checking if not primary because it still sets it to 89 on both

**Comment by NitroTF2** [comment]
> > > neither do you, since the angles are the same as everything else why check?
> > 
> > 
> > cuz works best to medic and I do change angle
> > based on my logics
> > 2022-08-10.23-08-30.mp4
> 
> tard, my point is that there's no point checking if not primary because it still sets it to 89 on both

sheeeeit my bad i am high

well it has my logic so it is good to good

**Comment by xxmyn** [comment]
ouch

**Pull request merged**